pkg/entrypoint/runner.go

```diff
@@ -300,9 +397,21 @@ func (rr *RealRunner) postLogMessage(ctx context.Context, mu *url.URL, request *
 }
 
 func (rr *RealRunner) putStatus(ctx context.Context, mu *url.URL, status *model.ActionStatus) error {
+	if err := os.MkdirAll(DefaultResultsPath, 0755); err == nil {
```

Comment:
When I rebase my testing changes on top of this, should I change this code path to cause "system" errors (i.e. return the errors) instead? Otherwise it seems like the Tekton conditions side won't necessarily line up with the files as-intended to be written, right?

Reply:
I would lean towards no for now, for a few specific reasons. At least not until we determine whether this use of Tekton when expressions is going to be permanent/expanded. Essentially (for now) I'd like the workflow to fully function using our conditions if the worst case, and only use the Tekton when expressions to shortcut for performance reasons. But that may need some investigation into what happens when the Tekton result is not available downstream, and how to ensure we fallback to our when condition logic instead.